sympy/printing/pretty/pretty_symbology.py

```diff
@@ -73,33 +58,32 @@ def pretty_use_unicode(flag=None):
 def pretty_try_use_unicode():
     """See if unicode output is available and leverage it if possible"""
 
-    try:
-        symbols = []
+    symbols = []
 
-        # see, if we can represent greek alphabet
-        symbols.extend(greek_unicode.values())
+    # see if we can represent greek alphabet
+    symbols += greek_unicode.values()
 
-        # and atoms
-        symbols += atoms_table.values()
+    # and atoms
+    symbols += atoms_table.values()
 
-        for s in symbols:
-            if s is None:
-                return  # common symbols not present!
+    for s in symbols:
+        if s is None:
+            return  # common symbols not present!
 
-            encoding = getattr(sys.stdout, 'encoding', None)
+        encoding = getattr(sys.stdout, 'encoding', None)
 
-            # this happens when e.g. stdout is redirected through a pipe, or is
-            # e.g. a cStringIO.StringO
-            if encoding is None:
-                return  # sys.stdout has no encoding
+        # this happens when e.g. stdout is redirected through a pipe, or is
+        # e.g. a cStringIO.StringO
+        if encoding is None:
+            return  # sys.stdout has no encoding
```

Comment:
I guess this could be at the top of the function.

Reply:
Oh yeah, doing this in the loop is pretty silly.

Reply:
Done